src/common/deploy-phase-common.ts

```diff
@@ -192,16 +192,17 @@ export function getResourceName(serviceContext: ServiceContext<ServiceConfig>) {
 }
 
 export function getTags(serviceContext: ServiceContext<ServiceConfig>): Tags {
-    let tags: Tags = {
-        app: serviceContext.appName,
-        env: serviceContext.environmentName
+    const serviceParams = serviceContext.params;
+
+    const handelTags: Tags = {
+             app: serviceContext.appName,
+             env: serviceContext.environmentName
     };
 
-    // Add user-defined tags if specified
-    const serviceParams = serviceContext.params;
-    if (serviceParams.tags) {
-        tags = Object.assign(tags, serviceParams.tags);
-    }
+    const appTags = serviceContext.tags;
+
+    const serviceTags = serviceParams.tags;
 
-    return tags;
+    // Service tags overwrite app tags, which overwrite Handel tags
+    return Object.assign({}, handelTags, appTags, serviceTags);
```

Comment:
I was about to ask why you were adding these when you haven't checked that they exist, but then I went and tried it out and it turns out Object.assign handles undefined values without a problem. You learn something new every day!

Reply:
Yup!  That was my TIL from a couple weeks ago :D